packages/omgidl-serialization/src/DeserializationInfoCache.ts

```diff
@@ -209,11 +238,95 @@ export class DeserializationInfoCache {
       name,
       type,
       typeDeserInfo: fieldDeserInfo,
+      isComplex: false,
       isArray,
       arrayLengths,
       definitionId: getDefinitionId(definition),
+      isOptional: isOptional(definition),
     };
   }
+
+  /** Returns default value for given FieldDeserializationInfo.
+   * If defaultValue is not defined on FieldDeserializationInfo, it will be calculated and set.
+   */
+  public getFieldDefault(deserInfo: FieldDeserializationInfo): unknown {
+    if (deserInfo.defaultValue != undefined) {
+      return deserInfo.defaultValue;
+    }
+    const { isArray, arrayLengths, type, isComplex } = deserInfo;
+
+    if (isArray === true && arrayLengths == undefined) {
+      deserInfo.defaultValue = [];
+      return deserInfo.defaultValue;
+    }
+    let defaultValueGetter;
+    if (isComplex) {
+      defaultValueGetter = () => {
+        return this.#getComplexDeserInfoDefault(
+          deserInfo.typeDeserInfo as ComplexDeserializationInfo,
+        );
+      };
+    } else {
+      // fixed length arrays are filled with default values
+      defaultValueGetter = PRIMITIVE_DEFAULT_VALUE_GETTERS.get(type);
+      if (!defaultValueGetter) {
+        throw new Error(`Failed to find default value getter for type ${type}`);
+      }
+    }
+    // Used for fixed length arrays that may be nested
+    const needsNestedArray = isArray === true && arrayLengths != undefined;
+    deserInfo.defaultValue = needsNestedArray
+      ? makeNestedArray(defaultValueGetter, arrayLengths, 0)
+      : defaultValueGetter();
+    return deserInfo.defaultValue;
+  }
+
+  /** Computes and sets the default value on the complex deserialization info */
+  #getComplexDeserInfoDefault(deserInfo: ComplexDeserializationInfo): Record<string, unknown> {
+    // if `structuredClone` is part of the environment, use it to clone the default message
+    // want to avoid defaultValues having references to the same object
```

Comment:
in what cases would this be bad?

Reply:
If a message contains several instances of a single default value, and the user of the library mutates a single one of those values, then all of the instances will be changed because they point to the same reference. This also includes default values across messages read by this parser. 
We don't do this in studio,  and it is an added cost to instantiate a new instance of this value each time, but I think that's what users (us) of this library will expect and it will save time fighting against potential errors that could come from manipulating (accidentally or otherwise) these objects sharing the same reference.